src/Orleans.Runtime/Placement/GlobalRandomPlacementDirector.cs

```diff
@@ -0,0 +1,29 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Orleans.Runtime.MembershipService;
+
+namespace Orleans.Runtime.Placement
+{
+    internal class GlobalRandomPlacementDirector : IPlacementDirector
+    {
+        private readonly GlobalMembershipTableManager membershipTableManager;
+
+        public GlobalRandomPlacementDirector(GlobalMembershipTableManager membershipTableManager)
+        {
+            this.membershipTableManager = membershipTableManager;
+        }
+
+        public Task<SiloAddress> OnAddActivation(PlacementStrategy strategy, PlacementTarget target, IPlacementContext context)
+        {
+            var activeSilos = this.membershipTableManager.MembershipTableSnapshot.Entries.Where(e => e.Value.Status == SiloStatus.Active).Select(e => e.Key);
+
+            var selectedSilos = activeSilos.Where(s => !s.Equals(context.LocalSilo));
+            if(!selectedSilos.Any())
+            {
+                selectedSilos = activeSilos;
+            }
+
+            return Task.FromResult(selectedSilos.First());
```

Comment:
Should actually be randomized no? Otherwise at least on a single silo this will likely be consistent, likely even across silos since they are reading the same table which should have the same order